eZ/Publish/Core/FieldType/Image/ImageStorage.php

```diff
@@ -113,6 +113,10 @@ public function storeFieldData(VersionInfo $versionInfo, Field $field, array $co
                 return false;
             }
 
+            if (!isset($field->value->data['id'])) {
```

Comment:
@tomaszmadeyski you were right, `$field->value->data` should be `null` when image is empty. However problem exists in a different place. See #2016.